Library/Homebrew/utils/analytics.rb

```diff
@@ -0,0 +1,54 @@
+
+def analytics_anonymous_prefix_and_os
+  @analytics_anonymous_prefix_and_os ||= begin
+    "#{OS_VERSION}, #{HOMEBREW_PREFIX.to_s.gsub(ENV["HOME"], "~")}"
+  end
+end
+
+def report_analytics(type, metadata={})
+  return unless ENV["HOMEBREW_ANALYTICS"]
+
+  metadata_args = metadata.map do |key, value|
+    ["-d", "#{key}=#{value}"] if key && value
+  end.compact.flatten
+
+  # Send analytics. Anonymise the IP address (aip=1) and don't send or store
+  # any personally identifiable information.
+  # https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide
+  # https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+  system "curl", "https://www.google-analytics.com/collect", "-d", "v=1",
+    "--silent", "--max-time", "3", "--output", "/dev/null",
+    "--user-agent", "#{HOMEBREW_USER_AGENT}",
+    "-d", "tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}",
+    "-d", "cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}",
+    "-d", "aip=1",
+    "-d", "an=Homebrew",
+    "-d", "av=#{HOMEBREW_VERSION}",
+    "-d", "t=#{type}",
+    *metadata_args
+end
+
+def report_analytics_event(category, action, label=analytics_anonymous_prefix_and_os, value=nil)
+  report_analytics(:event, {
+    :ec => category,
+    :ea => action,
+    :el => label,
+    :ev => value,
+  })
+end
+
+def report_analytics_exception(exception, options={})
+  if exception.is_a? BuildError
+    report_analytics_event("BuildError", e.formula.full_name)
+  end
```

Comment:
I think you should just send one request here. For two reason:
* it will benefit the performance.
* the second BuildError exception record is redundant and useless to us as it's already covered by the first event record.